src/agent/debuglet.ts

```diff
@@ -262,6 +262,24 @@ export class Debuglet extends EventEmitter {
         return;
       }
 
+      if (onGCP &&
+          (!that.config_.serviceContext ||
+           !that.config_.serviceContext.service)) {
```

Comment:
If the user specifies the empty string as the `service` we will still treat it as unspecified.  However, I don't think that is necessarily an issue.